File: 1422.c
int max(int i, int j) { return (i > j) ? i : j; }

int maxScore(char* c) {
    int n = strlen(c), r = 0, l = (c[0] == '0' ? 1 : 0);
    for (int i = 1; i < n; i++)
        if (c[i] == '1')
            ++r;
    int maxs = r + l;
    --n;
    for (int i = 1; i < n; i++)
        if (c[i] == '0') {
            ++l;
            maxs = max(maxs, r + l);
        } else
            --r;
    return maxs;
}

File: 1422.py
class Solution:
    def maxScore(self, c: str) -> int:
        n = len(c)
        r = 0
        l = 0
        if c[0] == "0":
            l = 1
        for i in range(1, n):
            if c[i] == "1":
                r += 1
        maxs = r + l
        for i in range(1, n - 1):
            if c[i] == "0":
                l += 1
                maxs = max(maxs, r + l)
            else:
                r -= 1
        return maxs
